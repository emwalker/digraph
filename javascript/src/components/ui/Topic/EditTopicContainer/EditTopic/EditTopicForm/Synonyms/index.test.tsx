import React from 'react'
import { shallow } from 'enzyme'

import { UnwrappedSynonyms } from './index'

const synonymEdge = {
  node: {
    name: 'Gnusto',
    id: '1234',
  },
}

const topic = {
  viewerCanDeleteSynonyms: true,
  synonyms: {
    edges: [synonymEdge],
  },
}

const props = {
  topic,
}

describe('<Synonyms />', () => {
  // @ts-expect-error
  const wrapper = shallow(<UnwrappedSynonyms {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
