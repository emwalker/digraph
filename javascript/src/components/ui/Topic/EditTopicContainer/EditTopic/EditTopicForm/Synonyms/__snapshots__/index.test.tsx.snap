// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Synonyms /> renders 1`] = `
<dl
  className="form-group"
>
  <label
    htmlFor="names-and-synonyms"
  >
    Names and synonyms
  </label>
  <ul
    className="Box list-style-none mt-1 mb-2"
  >
    <SynonymList
      onDelete={[Function]}
      onUpdate={[Function]}
      synonyms={
        Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": "1234",
                "name": "Gnusto",
              },
            },
          ],
        }
      }
    />
  </ul>
</dl>
`;
